### ISSUE TITLE
SideNav should be swippable (by default), so swipe left/right will close it

### ISSUE THREAD

**Issue by yoav-zibin** [angular/material#8310]
Compare the SideNav and the BottomSheet:
https://material.angularjs.org/latest/demo/sidenav
https://material.angularjs.org/latest/demo/bottomSheet

BottomSheet can be swiped down (to close it),
but SideNav can't be swiped left/right (to close it).

In all the google apps you can swipe left the left-nav to close it,
so that should be the default behavior.

Note that the SideNav can be locked-open, e.g., 
md-is-locked-open="$mdMedia('gt-xs')"
and when it's locked open then it shouldn't be swippable.

That's why I don't have an easy workaround, because adding:
md-swipe-left="onSwipeLeft()"
will always make it swippable.
Any suggestions for a workaround?

(BTW, the swipe demo on https://material.angularjs.org/latest/demo/swipe doesn't work for me, neither on desktop nor with in chrome mobile emulation.)

**Comment by mckenzielong** [comment]
Might be interested following in @DevVersion's PR: https://github.com/angular/material/pull/6174

**Comment by DevVersion** [comment]
Duplicate of #1591.

FYI: The PR is blocked by another PR.
